django/middleware/csrf.py

```diff
@@ -63,19 +99,35 @@ def rotate_token(request):
     """
     request.META.update({
         "CSRF_COOKIE_USED": True,
-        "CSRF_COOKIE": _get_new_csrf_key(),
+        "CSRF_COOKIE": _get_new_csrf_token(),
     })
+    request.csrf_cookie_needs_reset = True
 
 
 def _sanitize_token(token):
     # Allow only alphanum
-    if len(token) > CSRF_KEY_LENGTH:
-        return _get_new_csrf_key()
-    token = re.sub('[^a-zA-Z0-9]+', '', force_text(token))
-    if token == "":
-        # In case the cookie has been truncated to nothing at some point.
-        return _get_new_csrf_key()
-    return token
+    if re.search('[^a-zA-Z0-9]', force_text(token)):
+        return None
+    elif len(token) == CSRF_TOKEN_LENGTH:
+        return token
+    elif len(token) == CSRF_NONCE_LENGTH:
+        # For backwards compatibility, accept such values as unpadded nonces.
```

Comment:
I personally generate the NONCE's on the client side using javascript (so I don't need the token in the response body). It would be nice to continue long term to accept the raw cookie value as a valid csrf token.